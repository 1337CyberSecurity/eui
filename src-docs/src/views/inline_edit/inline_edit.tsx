import React from 'react';

import { EuiInlineEdit, EuiSpacer } from '../../../../src';

export default () => (
  <>
    <p>EuiInlineEdit - Text</p>
    <EuiInlineEdit
      display="text"
      inputAriaLabel="textControlInput"
      defaultValue="Hello World!"
    />

    <EuiSpacer />

    <p>EuiInlineEdit - Title</p>
    <EuiInlineEdit
      display="title"
      inputAriaLabel="textControlInput"
      defaultValue="Hello World (but as a title)!"
    />
  </>
);
