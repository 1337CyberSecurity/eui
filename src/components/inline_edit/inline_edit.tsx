/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, useState } from 'react';
import { CommonProps, ExclusiveUnion } from '../common';
import classNames from 'classnames';
import { EuiButtonEmpty, EuiButton } from '../button';
import { EuiFieldText, EuiFormRow, EuiForm } from '../form';
import { EuiTitle, EuiTitleSize } from '../title';
import { EuiText, EuiTextProps } from '../text';
import { EuiFlexGroup, EuiFlexItem } from '../flex';
// import { useEuiTheme } from '../../services';
// import { euiInlineEditStyles } from './inline_edit.styles';
import { htmlIdGenerator } from '../../services/accessibility';
import { useEuiI18n } from '../i18n';

export const DISPLAY_TYPES = ['title', 'text'] as const;
export type EuiInlineEditDisplayType = typeof DISPLAY_TYPES[number];

interface TitleDisplayProps {
  display: 'title';
  size?: EuiTitleSize;
}

interface TextDisplayProps {
  display: 'text';
  size?: EuiTextProps['size'];
}

export type EuiInlineEditProps = CommonProps &
  ExclusiveUnion<TextDisplayProps, TitleDisplayProps> & {
    /**
     * Display type for the text in readView
     */
    display: EuiInlineEditDisplayType;
    /**
     * Default string value for input in readView
     */
    defaultValue: string;
    /**
     * Allow users to pass in a function when the confirm button is clicked
     */
    onConfirm?: () => void;
    /**
     * Form label that appears above the form control
     * This is required for accessibility because there is no visual label on the input
     */
    inputAriaLabel: String;
    saveButtonAriaLabel?: string;
    cancelButtonAriaLabel?: string;
    /**
     * Start in editView
     */
    startWithEditOpen?: boolean;
  };

export const EuiInlineEdit: FunctionComponent<EuiInlineEditProps> = ({
  children,
  className,
  display,
  // m is the default for both EuiTitle and EuiText
  size = 'm',
  defaultValue,
  onConfirm,
  inputAriaLabel,
  saveButtonAriaLabel,
  cancelButtonAriaLabel,
  startWithEditOpen,
  ...rest
}) => {
  const classes = classNames('euiInlineEdit', className);

  // Styles to come later
  /*const theme = useEuiTheme();
  const styles = euiInlineEditStyles(theme);
  const cssStyles = [styles.euiInlineEdit];*/

  const [isInEdit, setIsInEdit] = useState(startWithEditOpen);
  const inlineEditInputId = htmlIdGenerator('__inlineEditInput')();

  /* Text Controls */
  const [editViewValue, setEditViewValue] = useState(defaultValue);
  const [readViewValue, setReadViewValue] = useState(defaultValue);

  const editTextViewOnChange = (e: any) => {
    setEditViewValue(e.target.value);
  };

  const saveTextEditValue = () => {
    const input = (document.getElementById(
      inlineEditInputId
    ) as HTMLInputElement).value;

    // If there's no text, cancel the action, reset the input text, and return to readView
    if (input) {
      setReadViewValue(input);
      setIsInEdit(!isInEdit);
      onConfirm && onConfirm();
    } else {
      setEditViewValue(readViewValue);
      setIsInEdit(!isInEdit);
    }
  };

  const defaultSaveButtonAriaLabel = useEuiI18n(
    'euiInlineEdit.saveButtonAriaLabel',
    'Save'
  );

  const defaultCancelButtonAriaLabel = useEuiI18n(
    'euiInlineEdit.cancelButtonAriaLabel',
    'Save'
  );

  const textEditViewElement = (
    <EuiForm fullWidth>
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiFieldText
            id={inlineEditInputId}
            value={editViewValue}
            onChange={editTextViewOnChange}
            aria-label={inputAriaLabel}
            autoFocus
            {...(rest as any)}
          />
        </EuiFlexItem>

        <EuiFlexItem grow={false}>
          <EuiFormRow>
            <EuiButton
              iconType="check"
              aria-label={saveButtonAriaLabel || defaultSaveButtonAriaLabel}
              onClick={saveTextEditValue}
              color="primary"
              fill
            >
              Save
            </EuiButton>
          </EuiFormRow>
        </EuiFlexItem>

        <EuiFlexItem grow={false}>
          <EuiFormRow>
            <EuiButton
              iconType="cross"
              aria-label={cancelButtonAriaLabel || defaultCancelButtonAriaLabel}
              onClick={() => {
                setEditViewValue(readViewValue);
                setIsInEdit(!isInEdit);
              }}
              color="primary"
            >
              Cancel
            </EuiButton>
          </EuiFormRow>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiForm>
  );

  const textReadViewElement = (
    <EuiButtonEmpty
      color="text"
      iconType="pencil"
      iconSide="right"
      autoFocus
      onClick={() => {
        setIsInEdit(!isInEdit);
      }}
    >
      {display === 'title' ? (
        <EuiTitle size={size as EuiTitleSize}>
          <h2>{readViewValue}</h2>
        </EuiTitle>
      ) : (
        <EuiText size={size as TextSize}>{readViewValue}</EuiText>
      )}
    </EuiButtonEmpty>
  );

  /* Current Form Control in View */
  const currentFormControlInView = isInEdit
    ? textEditViewElement
    : textReadViewElement;

  return (
    <div className={classes} {...rest}>
      {currentFormControlInView}
    </div>
  );
};
