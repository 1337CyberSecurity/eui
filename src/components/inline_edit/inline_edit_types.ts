/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { EuiButtonEmptyPropsForButton } from '../button/button_empty/button_empty';
import { EuiFieldTextProps } from '../form';

type _ButtonPropsWithoutOnClick = Omit<EuiButtonEmptyPropsForButton, 'onClick'>;

export type EuiInlineEditCommonProps = {
  defaultValue: string;
  /**
   * Allow users to pass in a function that is called when the confirm button is clicked
   * The function should return a boolean flag that will determine if the value will be saved.
   * When the flag is true, the value will be saved. When the flag is false, the user will be
   * returned to editMode.
   */
  onConfirm?: () => boolean;
  /**
   * Form label that appears above the form control
   * This is required for accessibility because there is no visual label on the input
   */
  inputAriaLabel: string;
  /**
   * Aria-label for save button in editMode
   */
  saveButtonAriaLabel?: string;
  /**
   * Aria-label for cancel button in editMode
   */
  cancelButtonAriaLabel?: string;
  /**
   * Start in editMode
   */
  startWithEditOpen?: boolean;
  /**
   * Props that will be applied directly to the EuiEmptyButton displayed in readMode
   */
  readModeProps?: _ButtonPropsWithoutOnClick;
  /**
   * Props that will be applied directly to the EuiFieldText displayed in editMode
   */
  editModeProps?: EuiFieldTextProps;
};
