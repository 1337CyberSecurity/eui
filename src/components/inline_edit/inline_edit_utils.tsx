/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { EuiButtonEmptyProps } from '../button';
import { EuiTextProps } from '../text';
import { EuiTitleSize } from '../title';

type InlineEditButtonSettings = {
  iconSize: EuiButtonEmptyProps['iconSize'];
  compressed: boolean;
};

/**
 *
 * @param size any value within EuiTextProps['size'] or EuiTitleSize
 * @returns returns the iconSize and compression combination to apply to buttons based on the text or title size provided.
 */
export const getInlineEditIconButtonSettings = (
  size: EuiTextProps['size'] | EuiTitleSize
) => {
  switch (size) {
    case 'xxxs':
    case 'xxs':
    case 'xs':
    case 's':
      return { iconSize: 's', compressed: true } as InlineEditButtonSettings;
    default:
      return { iconSize: 'm', compressed: false } as InlineEditButtonSettings;
  }
};
