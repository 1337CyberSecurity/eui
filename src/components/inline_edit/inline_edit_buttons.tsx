/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent } from 'react';
import { CommonProps } from '../common';
import classNames from 'classnames';
import { EuiButtonIcon } from '../button';
import { EuiFormRow } from '../form';
import { EuiTextProps } from '../text';
import { EuiTitleSize } from '../title';
import { EuiFlexItem } from '../flex';
import { useEuiI18n } from '../i18n';
import { getInlineEditIconButtonSettings } from './inline_edit_utils';

export type EuiInlineEditButtonsProps = CommonProps & {
  size?: EuiTextProps['size'] | EuiTitleSize;
  saveButtonAriaLabel?: string;
  cancelButtonAriaLabel?: string;
  saveFunction: Function;
  cancelFunction: Function;
};

export const EuiInlineEditButtons: FunctionComponent<EuiInlineEditButtonsProps> = ({
  className,
  size = 'm',
  saveButtonAriaLabel,
  cancelButtonAriaLabel,
  saveFunction,
  cancelFunction,
}) => {
  const classes = classNames('euiInlineEditButtons', className);

  const defaultSaveButtonAriaLabel = useEuiI18n(
    'euiInlineEditButtons.saveButtonAriaLabel',
    'Save edit'
  );

  const defaultCancelButtonAriaLabel = useEuiI18n(
    'euiInlineEditButtons.cancelButtonAriaLabel',
    'Cancel edit'
  );

  const buttonSettings = getInlineEditIconButtonSettings(size);

  return (
    <>
      <EuiFlexItem grow={false} className={classes}>
        <EuiFormRow>
          <EuiButtonIcon
            iconType="check"
            aria-label={saveButtonAriaLabel || defaultSaveButtonAriaLabel}
            onClick={() => saveFunction()}
            color="success"
            display="base"
            size={buttonSettings.compressed ? 's' : 'm'}
            iconSize={buttonSettings.iconSize}
          />
        </EuiFormRow>
      </EuiFlexItem>

      <EuiFlexItem grow={false}>
        <EuiFormRow>
          <EuiButtonIcon
            iconType="cross"
            aria-label={cancelButtonAriaLabel || defaultCancelButtonAriaLabel}
            onClick={() => cancelFunction()}
            color="danger"
            display="base"
            size={buttonSettings.compressed ? 's' : 'm'}
            iconSize={buttonSettings.iconSize}
          />
        </EuiFormRow>
      </EuiFlexItem>
    </>
  );
};
