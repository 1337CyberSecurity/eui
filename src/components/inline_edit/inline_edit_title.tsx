/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, useState } from 'react';
import { CommonProps } from '../common';
import classNames from 'classnames';
import { EuiButtonEmpty, EuiButtonEmptyProps } from '../button';
import { EuiFieldText, EuiForm } from '../form';
import { EuiTitle, EuiTitleSize } from '../title';
import { EuiFlexGroup, EuiFlexItem } from '../flex';
import { useGeneratedHtmlId } from '../../services/accessibility';
import { EuiInlineEditButtons } from './inline_edit_buttons';
import { getInlineEditIconButtonSettings } from './inline_edit_utils';
import { EuiInlineEditCommonProps } from './inline_edit_types';
import { EuiEmptyPromptProps } from '../empty_prompt';
// import { useEuiTheme } from '../../services';
// import { euiInlineEditStyles } from './inline_edit.styles';

export const HEADINGS = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const;
type Heading = typeof HEADINGS[number];

export type EuiInlineEditTitleProps = CommonProps &
  EuiInlineEditCommonProps & {
    /**
     * Title size level
     */
    size?: EuiTitleSize;
    /**
     * Level of heading to be used for the title
     */
    heading: Heading;
    /**
     * Default string value for input in readMode
     */
  };

export const EuiInlineEditTitle: FunctionComponent<EuiInlineEditTitleProps> = ({
  children,
  className,
  size = 'm',
  heading,
  defaultValue,
  onConfirm,
  inputAriaLabel,
  saveButtonAriaLabel,
  cancelButtonAriaLabel,
  startWithEditOpen,
  readModeProps,
  editModeProps,
  ...rest
}) => {
  const classes = classNames('euiInlineEditTitle', className);

  // Styles to come later
  /*const theme = useEuiTheme();
  const styles = euiInlineEditStyles(theme);
  const cssStyles = [styles.euiInlineEdit];*/

  const [isInEdit, setIsInEdit] = useState(startWithEditOpen);
  const inlineEditInputId = useGeneratedHtmlId({ prefix: '__inlineEditInput' });

  const [editModeValue, setEditModeValue] = useState(defaultValue);
  const [readModeValue, setReadModeValue] = useState(defaultValue);

  const saveTitleEditValue = () => {
    // If there's no text, cancel the action, reset the input text, and return to readMode
    if (editModeValue) {
      setReadModeValue(editModeValue);
      setIsInEdit(!isInEdit);
      onConfirm && onConfirm();
    } else {
      setEditModeValue(readModeValue);
      setIsInEdit(!isInEdit);
    }
  };

  const cancelTitleEdit = () => {
    setEditModeValue(readModeValue);
    setIsInEdit(!isInEdit);
  };

  const buttonSettings = getInlineEditIconButtonSettings(size);

  const titleEditModeElement = (
    <EuiForm fullWidth>
      <EuiFlexGroup gutterSize="s">
        <EuiFlexItem>
          <EuiFieldText
            id={inlineEditInputId}
            value={editModeValue}
            onChange={(e) => {
              setEditModeValue(e.target.value);
            }}
            aria-label={inputAriaLabel}
            autoFocus
            compressed={buttonSettings.compressed}
            {...editModeProps}
          />
        </EuiFlexItem>

        <EuiInlineEditButtons
          size={size as EuiTitleSize}
          saveFunction={saveTitleEditValue}
          cancelFunction={cancelTitleEdit}
        />
      </EuiFlexGroup>
    </EuiForm>
  );

  const H: Heading = heading;
  const titleReadModeElement = (
    <EuiButtonEmpty
      color="text"
      iconType="pencil"
      iconSide="right"
      autoFocus
      flush="both"
      iconSize={buttonSettings.iconSize}
      size={buttonSettings.compressed ? 's' : 'm'}
      onClick={() => {
        setIsInEdit(!isInEdit);
      }}
      {...readModeProps}
    >
      <EuiTitle size={size}>
        <H>{readModeValue}</H>
      </EuiTitle>
    </EuiButtonEmpty>
  );

  return (
    <div className={classes} {...rest}>
      {isInEdit ? titleEditModeElement : titleReadModeElement}
    </div>
  );
};
