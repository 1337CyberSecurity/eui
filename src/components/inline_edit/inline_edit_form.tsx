/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, useState } from 'react';
import classNames from 'classnames';
import { CommonProps } from '../common';
import { EuiFieldTextProps } from '../form';
import { EuiButtonEmpty, EuiButtonEmptyProps } from '../button';
import { EuiButtonEmptyPropsForButton } from '../button/button_empty/button_empty';
import { EuiFlexGroup, EuiFlexItem } from '../flex';
import { useGeneratedHtmlId } from '../../services/accessibility';
import { EuiInlineEditButtons } from './inline_edit_buttons';
import { getInlineEditIconButtonSettings } from './inline_edit_utils';
import { EuiInlineEditTitleProps } from './inline_edit_title';
import { EuiInlineEditTextProps } from './inline_edit_text';
// import { useEuiTheme } from '../../services';
// import { euiInlineEditStyles } from './inline_edit.styles';

type _ButtonPropsWithoutOnClick = Omit<EuiButtonEmptyPropsForButton, 'onClick'>;

export type EuiInlineEditFormProps = CommonProps & {
  defaultValue: string;
  /**
   * Allow users to pass in a function that is called when the confirm button is clicked
   * The function should return a boolean flag that will determine if the value will be saved.
   * When the flag is true, the value will be saved. When the flag is false, the user will be
   * returned to editMode.
   */
  onConfirm?: () => boolean;
  /**
   * Form label that appears above the form control
   * This is required for accessibility because there is no visual label on the input
   */
  inputAriaLabel: string;
  /**
   * Aria-label for save button in editMode
   */
  saveButtonAriaLabel?: string;
  /**
   * Aria-label for cancel button in editMode
   */
  cancelButtonAriaLabel?: string;
  /**
   * Start in editMode
   */
  startWithEditOpen?: boolean;
  /**
   * Props that will be applied directly to the EuiEmptyButton displayed in readMode
   */
  readModeProps?: _ButtonPropsWithoutOnClick;
  /**
   * Props that will be applied directly to the EuiFieldText displayed in editMode
   */
  editModeProps?: EuiFieldTextProps;
};

export const EuiInlineEditForm: FunctionComponent<EuiInlineEditFormProps> = ({
  className,
  children,
  updateReadModeValue,
  props,
}) => {
  const classes = classNames('euiInlineEditForm', className);

  // Styles to come later! (Styling editMode text to match the size of its readMode counterpart)
  /*const theme = useEuiTheme();
  const styles = euiInlineEditStyles(theme);
  const cssStyles = [styles.euiInlineEdit];*/

  const {
    startWithEditOpen,
    size,
    defaultValue,
    onConfirm,
    inputAriaLabel,
    saveButtonAriaLabel,
    cancelButtonAriaLabel,
    readModeProps,
    editModeProps,
  } = props;

  const [isEditing, setIsEditing] = useState(false || startWithEditOpen);
  const inlineEditInputId = useGeneratedHtmlId({ prefix: '__inlineEditInput' });

  const [editModeValue, setEditModeValue] = useState(defaultValue);
  const [readModeValue, setReadModeValue] = useState(defaultValue);

  const cancelInlineEdit = () => {
    setEditModeValue(readModeValue);
    setIsEditing(!isEditing);
  };

  const saveInlineEditValue = () => {
    if (editModeValue && onConfirm && !onConfirm()) {
      // If there is text, an onConfirm method is present, and it has returned false, cancel the action
      return;
    } else if (editModeValue) {
      setReadModeValue(editModeValue);
      updateReadModeValue(editModeValue);
      setIsEditing(!isEditing);
    } else {
      // If there's no text, cancel the action, reset the input text, and return to readMode
      cancelInlineEdit();
    }
  };

  const buttonSettings = getInlineEditIconButtonSettings(size);

  const editModeForm = (
    <EuiForm fullWidth>
      <EuiFlexGroup gutterSize="s">
        <EuiFlexItem>
          <EuiFieldText
            id={inlineEditInputId}
            value={editModeValue}
            onChange={(e) => {
              setEditModeValue(e.target.value);
            }}
            aria-label={inputAriaLabel}
            autoFocus
            compressed={buttonSettings.compressed}
            {...editModeProps}
          />
        </EuiFlexItem>

        <EuiInlineEditButtons
          size={size}
          saveFunction={saveInlineEditValue}
          cancelFunction={cancelInlineEdit}
          saveButtonAriaLabel={saveButtonAriaLabel}
          cancelButtonAriaLabel={cancelButtonAriaLabel}
        />
      </EuiFlexGroup>
    </EuiForm>
  );

  const readModeElement = (
    <EuiButtonEmpty
      color="text"
      iconType="pencil"
      iconSide="right"
      autoFocus
      flush="both"
      iconSize={buttonSettings.iconSize}
      size={buttonSettings.compressed ? 's' : 'm'}
      onClick={() => {
        setIsEditing(!isEditing);
      }}
      {...readModeProps}
    >
      {children}
    </EuiButtonEmpty>
  );

  return (
    <div className={classes}>{isEditing ? editModeForm : readModeElement}</div>
  );
};
