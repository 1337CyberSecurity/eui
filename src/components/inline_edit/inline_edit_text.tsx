/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, useState } from 'react';
import classNames from 'classnames';
import { EuiText, EuiTextProps } from '../text';
import { EuiInlineEditForm, EuiInlineEditFormProps } from './inline_edit_form';

export type EuiInlineEditTextSizes = Exclude<EuiTextProps['size'], 'relative'>;

export type EuiInlineEditTextProps = EuiInlineEditFormProps & {
  /**
   * Text size level
   */
  size?: EuiInlineEditTextSizes;
};

export const EuiInlineEditText: FunctionComponent<EuiInlineEditTextProps> = ({
  children,
  className,
  size = 'm',
  defaultValue,
  onConfirm,
  inputAriaLabel,
  saveButtonAriaLabel,
  cancelButtonAriaLabel,
  startWithEditOpen,
  readModeProps,
  editModeProps,
  ...rest
}) => {
  const classes = classNames('euiInlineEditText', className);

  const [textReadModeValue, setTextReadModeValue] = useState(defaultValue);

  const formProps = {
    size,
    defaultValue,
    onConfirm,
    inputAriaLabel,
    saveButtonAriaLabel,
    cancelButtonAriaLabel,
    startWithEditOpen,
    readModeProps,
    editModeProps,
  };

  return (
    <div className={classes} {...rest}>
      <EuiInlineEditForm
        {...formProps}
        updateReadModeValue={setTextReadModeValue}
      >
        <EuiText size={size}>{textReadModeValue}</EuiText>
      </EuiInlineEditForm>
    </div>
  );
};
