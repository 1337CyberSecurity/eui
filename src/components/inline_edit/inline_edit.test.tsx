/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { EuiInlineEdit } from './inline_edit';

describe('EuiInlineEdit', () => {
  describe('props', () => {
    test('renders as text', () => {
      const { container } = render(
        <EuiInlineEdit
          display="text"
          inputAriaLabel={'textInput'}
          defaultValue="hello world"
          {...requiredProps}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test('renders as a title', () => {
      const component = render(
        <EuiInlineEdit
          display="text"
          inputAriaLabel={'titleInput'}
          defaultValue="hello world"
          {...requiredProps}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
