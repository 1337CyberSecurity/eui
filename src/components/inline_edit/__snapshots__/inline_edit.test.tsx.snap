// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiInlineEdit props renders as a title 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="aria-label"
        class="euiInlineEdit testClass1 testClass2"
        data-test-subj="test subject string"
      >
        <button
          class="euiButtonEmpty css-wvaqcf-empty-text"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
          >
            <span
              class="euiButtonContent__icon"
              color="inherit"
              data-euiicon-type="pencil"
            />
            <span
              class="euiButtonEmpty__text"
            >
              <div
                class="euiText emotion-euiText-m"
              >
                hello world
              </div>
            </span>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="aria-label"
      class="euiInlineEdit testClass1 testClass2"
      data-test-subj="test subject string"
    >
      <button
        class="euiButtonEmpty css-wvaqcf-empty-text"
        type="button"
      >
        <span
          class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
        >
          <span
            class="euiButtonContent__icon"
            color="inherit"
            data-euiicon-type="pencil"
          />
          <span
            class="euiButtonEmpty__text"
          >
            <div
              class="euiText emotion-euiText-m"
            >
              hello world
            </div>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByTestSubject": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByTestSubject": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByTestSubject": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByTestSubject": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByTestSubject": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByTestSubject": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`EuiInlineEdit props renders as text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="aria-label"
        class="euiInlineEdit testClass1 testClass2"
        data-test-subj="test subject string"
      >
        <button
          class="euiButtonEmpty css-wvaqcf-empty-text"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
          >
            <span
              class="euiButtonContent__icon"
              color="inherit"
              data-euiicon-type="pencil"
            />
            <span
              class="euiButtonEmpty__text"
            >
              <div
                class="euiText emotion-euiText-m"
              >
                hello world
              </div>
            </span>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="aria-label"
      class="euiInlineEdit testClass1 testClass2"
      data-test-subj="test subject string"
    >
      <button
        class="euiButtonEmpty css-wvaqcf-empty-text"
        type="button"
      >
        <span
          class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
        >
          <span
            class="euiButtonContent__icon"
            color="inherit"
            data-euiicon-type="pencil"
          />
          <span
            class="euiButtonEmpty__text"
          >
            <div
              class="euiText emotion-euiText-m"
            >
              hello world
            </div>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByTestSubject": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByTestSubject": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByTestSubject": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByTestSubject": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByTestSubject": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByTestSubject": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
